docs/how-to/use-autosave-in-an-ioc.rst

```diff
@@ -0,0 +1,65 @@
+Use `softioc.autosave` in an IOC
+=======================
+
+`../tutorials/creating-an-ioc` shows how to create a pythonSoftIOC.
+
+
+Example IOC
+-----------
+
+.. literalinclude:: ../examples/example_autosave_ioc.py
+
+Records are instantiated as normal and configured for automatic loading and
+periodic saving to a backup file with the keyword arguments ``autosave``, ``autosave_fields``
+and ``autosave_name``. Records with ``autosave=True`` (``False`` by default) have their
+VAL fields backed up. Additional record fields in a list passed to ``autosave_fields`` will be backed
+up, note that this applies even when ``autosave`` is ``False``.
+
+The field values get written into a yaml-formatted file containing key-value pairs,
+by default the keys are the same as the record name passed as the first argument to the
+`builder.<RECORD_TYPE>()` call, excluding the device name specified in the `builder.SetDeviceName()`
+call, but this can be changed to a string name specified in the ``autosave_name`` keyword
+argument.
```

Comment:
Why on earth would we want to do this?

Reply:
I personally wouldn't, by default we are stripping the device name from the autosave key to ensure consistency in the case where some PVs are instantiated before builder.SetDeviceName() and some are instantiated after. Users could manually set the autosave key to <device_name>.<pv_name> if they desired this, or any other naming scheme, though I suppose it's not necessary.

Reply:
Unless there's a compelling use case (and this doesn't sound like it) I'd drop support for this option.

This interaction with `SetDeviceName()` sounds a bit troubling; can the name of a PV change after it has been created?  I thought the record name was fixed at the time of creation (the call to `aIn()` etc)?

Reply:
As far as I know there's no public way to change the record suffix, but the name that gets passed to the ProcessDeviceSupportCore (and subclass etc) init is record.name, see RecordWraper.__init__()

```
        record = builder(name, **fields)
        record.address = '@' + record.name
        self.__set('__builder', record)
        self.__set('__device',  device(record.name, **device_kargs))
```
record.name will either be <device_name>.<name> or just <name> if the prefix is not set, I think this name would otherwise become meaningless after LoadDatabase() is called. autosave_name could be removed if we passed the RecordWrapper.__init__ kwarg name directly instead of record.name in the device(record.name, device_kargs) call, though I haven't tested if anything could possibly break from doing that.